File: modules/wem-api/src/main/java/com/enonic/wem/query/expr/NotExpr.java
package com.enonic.wem.query.expr;

public final class NotExpr
    implements ConstraintExpr
{
    private final Expression expr;

    public NotExpr( final ConstraintExpr expr )
    {
        this.expr = expr;
    }

    public Expression getExpression()
    {
        return this.expr;
    }

    @Override
    public String toString()
    {
        return "NOT (" + this.expr.toString() + ")";
    }
}

File: modules/admin-ui/src/main/resources/web/admin/common/js/query/expr/NotExpr.ts
module api.query.expr {

    export class NotExpr implements ConstraintExpr {
        private expr: Expression;

        constructor(expr: ConstraintExpr) {
            this.expr = expr;
        }

        getExpression(): Expression {
            return this.expr;
        }

        toString() {
            return "NOT (" + this.expr.toString() + ")";
        }
    }
}
